data_diff/cli_options.py

```diff
@@ -0,0 +1,48 @@
+from typing import Optional, Literal, Dict, Union, Tuple
+
+from pydantic import BaseModel, PositiveInt
+
+
+class CliOptions(BaseModel):
```

Comment:
This class is bloated and will become bloated over time, but I'm willing to accept that tradeoff for better runtime validation of flag types and simplification of `kwargs**` across method calls. Overall, this PR has opened my eyes that a major refactor needs to happen across lots of the codebase for simpler data flows, but in the short-term, this approach is worth it.